hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/pipes/Application.java

```diff
@@ -133,6 +135,10 @@
     
     process = runClient(cmd, env);
     clientSocket = serverSocket.accept();
+    // start ping socket cleaner
+    socketCleaner = new PingSocketCleaner("ping-socket-cleaner");
+    socketCleaner.setDaemon(true);
+    socketCleaner.start();
```

Comment:
The cleaner thread is never stopped. Would you clean up the thread in the cleanup method?

Reply:
ok, i will do that in cleanup method